File: java/src/main/java/com/alxarafe/app/application/catalogue/ItemRepository.java
package com.alxarafe.app.application.catalogue;

import com.alxarafe.app.domain.catalogue.entity.Item;
import com.alxarafe.app.domain.catalogue.valueobject.Sku;

import java.util.List;
import java.util.Optional;

public interface ItemRepository {
    Optional<Item> findBySku(Sku sku);

    List<ItemView> findAll();

    void save(Item item);
}

File: php/src/Application/Catalogue/ItemRepository.php
<?php

declare(strict_types=1);

namespace Alxarafe\App\Application\Catalogue;

use Alxarafe\App\Domain\Catalogue\Entity\Item;
use Alxarafe\App\Domain\Catalogue\ValueObject\Sku;

interface ItemRepository
{
    public function findBySku(Sku $sku): ?Item;

    /** @return list<ItemView> */
    public function findAll(): array;

    public function save(Item $item): void;
}
